library/aes.c

```diff
@@ -675,6 +682,31 @@ void mbedtls_aes_xts_free( mbedtls_aes_xts_context *ctx )
 }
 #endif /* MBEDTLS_CIPHER_MODE_XTS */
 
+static int mbedtls_generate_fake_key( unsigned int keybits, mbedtls_aes_context *ctx )
+{
+    unsigned int key_bytes = keybits >> 3;
+    int qword = (int) ( keybits >> 5 );
+
+    if( ctx->frk != NULL )
+    {
+        mbedtls_free( ctx->frk );
+        ctx->frk = NULL;
+    }
+
+    ctx->frk = mbedtls_calloc( 1, key_bytes );
```

Comment:
I think it will be better to define MAX_AES_KEY_SIZE value, in our case it will be 128 bit and add fake key as a buffer with MAX_KEY_SIZE length to context. It will dismiss  all alloc/free operations and checks

Reply:
Sure, we can do it this way.

Reply:
Thanks! I t seems I didn't expand the file :)